File: 프로그래머스/2/12900. 2 x n 타일링/2 x n 타일링.py
def solution(n):
    mod = 1_000_000_007
    dp = list(range(n + 1))

    for i in range(3, n + 1):
        dp[i] = (dp[i - 1] + dp[i - 2]) % mod
    
    return dp[n]

File: 프로그래머스/2/12900. 2 x n 타일링/2 x n 타일링.java
class Solution {
    
    private static final int MOD = 1_000_000_007;
    
    public int solution(int n) {        
        int[] dp = new int[n + 1];
        for (int i = 0; i <= n; i++) {
            dp[i] = i;
        }
        for (int i = 3; i <= n; i++) {
            dp[i] = (dp[i - 1] + dp[i - 2]) % MOD;
        }
        return dp[n];
    }
}